### PR TITLE
Don't start land checks if the PR hasn't been approved yet

### PR DESCRIPTION
**Pull request by huydhn**
Per title, don't start land checks if the PR hasn't been approved yet. This is very important to make sure that we don't start CI jobs from unknown devs, i.e. first time contributor.

Also rename force to `skip_mandatory_checks` to make it clearer on what this flag does

### Testing

```
python .github/scripts/test_trymerge.py
```

### CODE REVIEW

File: .github/scripts/trymerge.py
Comment (malfet) [inline .github/scripts/trymerge.py:1307]: Occams Razor: do not throw, but simply pass "force" flag to validate only reviews/CLA part of the rule
```suggestion
    # Important: check for merge rule once before starting land checks
    # because we want to make sure that only approved PRs can start CI
    # jobs. If there's missing approval, a RuntimeError will be raised
    # here to stop the merge process right away
    find_matching_merge_rule(pr, repo, force=True)
```

Comment (huydhn) [reply .github/scripts/trymerge.py:1307]: Ohhh, this is wicked. I didn't realize that the `force` flag behaves that way. But you're right, setting `force` to true will only include CLA check in the list of required checks and achieves what we need here.

File: .github/scripts/trymerge.py
Comment (zengk95) [inline .github/scripts/trymerge.py:1294]: land_check_commit will always be None here if we move this.

File: .github/scripts/trymerge.py
Comment (zengk95) [inline .github/scripts/trymerge.py:1302]: nit: feel like we should rename the force param cause we're not actually forcing it. Also can we move this down to the if land_checks: ? If we make changes to this function, we might unintentionally break force merge on lines 1304-1307

**Review by zengk95 — APPROVED** [review]
lgtm

File: .github/scripts/trymerge.py
Comment (ZainRizvi) [reply .github/scripts/trymerge.py:1294]: +1, also, it makes sense to post this comment _after_ we've done the first round of sanity checks to ensure we'll actually try to merge the pr

**Review by ZainRizvi — CHANGES_REQUESTED** [review]

**Comment by zengk95** [comment]
The faiure message looks a bit weird. Like it says "Reason: View failures on HUD". It's not really a reason, more of a trouble shooting.

**Review by malfet — CHANGES_REQUESTED** [review]
I thought I've commented on this yesterday, but apparently I did not: this change breaks the behavior of forced merges of stale changes. 

So, not to leave it to the mercy of the human reviewer, can we please add a test (as separate PR), that checks that force merges override staleness?

File: .github/scripts/trymerge.py
Comment (huydhn) [reply .github/scripts/trymerge.py:1296]: Interesting, so we want to allow force merge stale PRs.  When I move this block up, my intention was to not allow that anymore, so a stale PR would need to be rebased first before allowing to merge (force or not).  If you think that's a feature that we want to keep, I can put this back.

**Comment by ZainRizvi** [comment]
> The faiure message looks a bit weird. Like it says "Reason: View failures on HUD". It's not really a reason, more of a trouble shooting.

Yeah, I'm fixing that now

File: .github/scripts/trymerge.py
Comment (huydhn) [inline .github/scripts/trymerge.py:1307]: Note that `merge_into` already has a similar `find_matching_merge_rule` check, so PR without approvals won't run there neither

**Review by ZainRizvi — APPROVED** [review]

File: .github/scripts/trymerge.py
Comment (huydhn) [reply .github/scripts/trymerge.py:1330]: I guess we can discuss more on that in a separate PR/issue. I do agree with you though that stale check should be done first before force and land checks, but may be there are legit use cases to allow that

**Pull request closed without merging**
